nixos/modules/tasks/filesystems/zfs.nix

```
@@ -642,41 +642,14 @@ in
         };
 
         scriptArgs = "%i";
-        path = [ pkgs.gawk cfgZfs.package ];
-
-        # ZFS has no way of enumerating just devices in a pool in a way
-        # that 'zpool online -e' supports. Thus, we've implemented a
-        # bit of a strange approach of highlighting just devices.
-        # See: https://github.com/openzfs/zfs/issues/12505
-        script = let
-          # This UUID has been chosen at random and is to provide a
-          # collision-proof, predictable token to search for
-          magicIdentifier = "NIXOS-ZFS-ZPOOL-DEVICE-IDENTIFIER-37108bec-aff6-4b58-9e5e-53c7c9766f05";
-          zpoolScripts = pkgs.writeShellScriptBin "device-highlighter" ''
-            echo "${magicIdentifier}"
-          '';
-        in ''
+        path = [ pkgs.gawk cfgZfs.package pkgs.parted pkgs.cloud-utils pkgs.util-linux ];
```

(Missed this earlier)

Do we really need `cloud-utils` and `util-linux` in the path if we `substituteInPlace`d them in the actual `zpool-auto-expand-partitions` derivation?

Should we also substitute `parted` in that derivation?

---

```suggestion
        path = with pkgs; [ gawk cfgZfs.package parted cloud-utils util-linux ];
```
I always find this easier to parse the packages.